polyglot/piranha/src/models/matches.rs

```diff
@@ -41,5 +38,41 @@ impl Match {
   pub(crate) fn matches(&self) -> &HashMap<String, String> {
     &self.matches
   }
+}
+
+fn ser_range<S: Serializer>(range: &Range, serializer: S) -> Result<S::Ok, S::Error> {
```

Comment:
ok. I have had to do this because - originally `Range` is declared in the `tree-sitter` library but it doesn't have the `serialize` trait. So this is a trick I saw SO - basically cast it to a local struct that implements serialize and serialize this local stuct.